static/app/components/stateLayer.tsx

```diff
@@ -0,0 +1,67 @@
+import isPropValid from '@emotion/is-prop-valid';
+import styled from '@emotion/styled';
+
+import {defined} from 'sentry/utils';
+
+interface StateLayerProps extends React.HTMLAttributes<HTMLSpanElement> {
+  color?: string;
+  higherOpacity?: boolean;
+  isHovered?: boolean;
+  isPressed?: boolean;
+}
+
+const StateLayer = styled(
```

Comment:
I think this name could be a bit more descriptive... `InteractableStateLayer` maybe?

Reply:
Hmm good point actually. Named it `InteractionStateLayer`, as "Interaction State" is a commonly used term in design systems.